pkg/api/v1alpha1/cluster_webhook.go

```diff
@@ -219,3 +225,31 @@ func (r *Cluster) ValidateDelete() error {
 
 	return nil
 }
+
+func buildStatusError(qualifiedKind schema.GroupKind, name string, errs []error) *apierrors.StatusError {
+	causes := make([]metav1.StatusCause, 0, len(errs))
+	for i := range errs {
+		err := errs[i]
```

Comment:
any reason to not get the element from the range directly?
```suggestion
	for _, err := range errs {
```